command/acl_auth_method_create.go

```diff
@@ -0,0 +1,174 @@
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault|usageOptsNoNamespace) + `
+
+ACL Auth Method Create Options:
+
+  -name
+    Sets the human readable name for the ACL auth method. The name must be
+    between 1-128 characters and is a required parameter.
+
+  -type
+    Sets the type of the auth method. Currently the only supported type is
+    'OIDC'.
+
+  -max-token-ttl
+    Sets the duration of time all tokens created by this auth method should be
+    valid for.
+
+  -token-locality
+    Defines the kind of token that this auth method should produce. This can be
+    either 'local' or 'global'. If empty the value of 'local' is assumed.
+
+  -default
+    Specifies whether this auth method should be treated as a default one in
+    case no auth method is explicitly specified for a login command.
+
+  -config
+    Auth method configuration in JSON format.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(a.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-name":           complete.PredictAnything,
+			"-type":           complete.PredictAnything,
+			"-max-token-ttl":  complete.PredictAnything,
+			"-token-locality": complete.PredictAnything,
+			"-default":        complete.PredictAnything,
+			"-config":         complete.PredictNothing,
+		})
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (a *ACLAuthMethodCreateCommand) Synopsis() string { return "Create a new ACL auth method" }
+
+// Name returns the name of this command.
+func (a *ACLAuthMethodCreateCommand) Name() string { return "acl auth-method create" }
+
+// Run satisfies the cli.Command Run function.
+func (a *ACLAuthMethodCreateCommand) Run(args []string) int {
+
+	flags := a.Meta.FlagSet(a.Name(), FlagSetClient)
+	flags.Usage = func() { a.Ui.Output(a.Help()) }
+	flags.StringVar(&a.name, "name", "", "")
+	flags.StringVar(&a.methodType, "type", "", "")
+	flags.StringVar(&a.tokenLocality, "token-locality", "local", "")
+	flags.DurationVar(&a.maxTokenTTL, "max-token-ttl", 0, "")
+	flags.BoolVar(&a.isDefault, "default", true, "")
+	flags.StringVar(&a.config, "config", "", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments.
+	if len(flags.Args()) != 0 {
+		a.Ui.Error("This command takes no arguments")
+		a.Ui.Error(commandErrorText(a))
+		return 1
+	}
+
+	// Perform some basic validation
+	if a.name == "" {
+		a.Ui.Error("ACL auth method name must be specified using the -name flag")
+		return 1
+	}
+	if !slices.Contains([]string{"global", "local"}, a.tokenLocality) {
+		a.Ui.Error("Token locality must be set to either 'local' or 'global'")
+		return 1
+	}
+	if strings.ToLower(a.methodType) != "oidc" {
```

Comment:
In the RPC we check for `OIDC` within validation, so I wonder here if it would be worth using `ToUpper` and writing that value back on the config? I am thinking of cases where one or more of the letters is lower case, which would pass validation here but fail within the RPC.

Reply:
Fair enough, we should be consistent.